src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs

```diff
@@ -358,247 +359,236 @@ private static object[] BindArguments(byte[] input, ref int offset, IReadOnlyLis
         /// <inheritdoc />
         public void WriteMessage(HubMessage message, IBufferWriter<byte> output)
         {
-            var writer = MemoryBufferWriter.Get();
+            var writer = new MessagePackWriter(output);
 
-            try
-            {
-                // Write message to a buffer so we can get its length
-                WriteMessageCore(message, writer);
+            // Write message to a buffer so we can get its length
+            WriteMessageCore(message, ref writer);
 
-                // Write length then message to output
-                BinaryMessageFormatter.WriteLengthPrefix(writer.Length, output);
-                writer.CopyTo(output);
-            }
-            finally
-            {
-                MemoryBufferWriter.Return(writer);
-            }
+            // REVIEW : I have no idea if this is needed or not, the previous code was using memryBufferWritter.CopyTo(stream)
+            //writer.Flush();
```

Comment:
Yes, you should absolutely flush `MessagePackWriter` before assuming that all bytes have been written to the underlying `IBufferWriter<byte>`.